sandbox/lua/filters/procstat.lua

```diff
@@ -0,0 +1,191 @@
+
+--[[
+
+Calculates deltas in /proc/stat data.
+Also emits CPU percentage utilization information.
+
+Config:
+
+- sec_per_row (uint, optional, default 60)
+    Sets the size of each bucket (resolution in seconds) in the sliding window.
+
+- whitelist_pattern (string)
+    Only proccess fields the fit the pattern, defaults to match the global cpu: "cpu$"
+    see: http://lua-users.org/wiki/PatternsTutorial
+
+*Example Heka Configuration*
+
+.. code-block:: ini
+
+    [ProcStats]
+    type = "FilePollingInput"
+    ticker_interval = 1
+    file_path = "/proc/stat"
+    decoder = "ProcStatDecoder"
+
+    [ProcStatDecoder]
+    type = "SandboxDecoder"
+    filename = "lua_decoders/linux_procstat.lua"
+
+    [ProcStatFilter]
+    type = "SandboxFilter"
+    filename = "lua_filters/procstat.lua"
+    preserve_data = true
+    message_matcher = "Type == 'stats.procstat'"
+    [ProcStatFilter.config]
+        whitelist_regex = "cpu*"
+
+
+Cpu fields:
+    1    2    3      4    5        6     7         8       9       10
+    user nice system idle [iowait] [irq] [softirq] [steal] [guest] [guestnice]
+    Note: systems provide user, nice, system, idle. Other fields depend on kernel.
+
+    user: Time spent executing user applications (user mode).
+    nice: Time spent executing user applications with low priority (nice).
+    system: Time spent executing system calls (system mode).
+    idle: Idle time.
+    iowait: Time waiting for I/O operations to complete.
+    irq: Time spent servicing interrupts.
+    softirq: Time spent servicing soft-interrupts.
+    steal: ticks spent executing other virtual hosts [virtualization setups]
+    guest: Used in virtualization setups.
+    guestnice: running a niced guest
+
+intr
+    This line shows counts of interrupts serviced since
+    boot time, for each of the possible system interrupts.
+    The first column is the total of all interrupts
+    serviced including unnumbered architecture specific
+    interrupts; each subsequent column is the total for
+    that particular numbered interrupt.  Unnumbered
+    interrupts are not shown, only summed into the total.
+
+ctxt 115315
+    The number of context switches that the system
+    underwent.
+
+btime 769041601
+    boot time, in seconds since the Epoch, 1970-01-01
+    00:00:00 +0000 (UTC).
+
+processes 86031
+    Number of forks since boot.
+
+procs_running 6
+    Number of processes in runnable state.  (Linux 2.5.45 onward.)
+
+procs_blocked 2
+    Number of processes blocked waiting for I/O to complete.  (Linux 2.5.45 onward.)
+
+softirq 288977 23 101952 19 13046 19217 7 19125 92077 389 43122
+    Time spent servicing soft-interrupts.
+--]]
+
+require 'string'
+
+local proc_stat_previous = nil
+local proc_stat_mappings = {
+    cpu = {
+        'user', 'nice', 'system', 'idle', 'iowait', 'irq',
+        'softirq', 'steal', 'guest', 'guestnice'
+    }
+}
+
+local preserve_data = read_config('preserve_data')
```

Comment:
i think i had it heare as a note because the previous developer was saving or passing the Payload on...
I think ill drop it.